service-base/src/main/java/org/eclipse/hono/service/management/device/DeviceManagementHttpEndpoint.java

```diff
@@ -100,29 +113,42 @@ public void addRoutes(final Router router) {
                 .handler(this::doDeleteDevice);
     }
 
+    /**
+     * The service to forward requests to.
+     *
+     * @return The service to bind to, must never return {@code null}.
+     */
+    protected abstract DeviceManagementService getService();
+
     private void doGetDevice(final RoutingContext ctx) {
 
         final String deviceId = getDeviceIdParam(ctx);
         final String tenantId = getTenantParam(ctx);
         final HttpServerResponse response = ctx.response();
 
-        final JsonObject requestMsg = EventBusMessage.forOperation(RegistryManagementConstants.ACTION_GET)
-                .setTenant(tenantId)
-                .setDeviceId(deviceId)
-                .toJson();
+        final Span span = Util.newChildSpan(SPAN_NAME_GET_DEVICE, TracingHandler.serverSpanContext(ctx), tracer, tenantId, deviceId, getClass().getSimpleName());
+
+        if (tenantId == null || deviceId == null) {
+            HttpUtils.badRequest(ctx, "Missing tenant or device id");
```

Comment:
shouldn't the processing of the request end here?

Reply:
Yes. Good catch that was missing in the old code as well. The request ends by badReqeust() call, but we should return immediatelly.